server/src-lib/Hasura/Server/Auth/JWT.hs

```diff
@@ -148,8 +149,10 @@ updateJwkRef (Logger logger) manager url jwkRef = do
     logAndThrow err
 
   let parseErr e = JFEJwkParseError (T.pack e) $ "Error parsing JWK from url: " <> urlT
-  jwkset <- either (logAndThrow . parseErr) return $ J.eitherDecode respBody
-  liftIO $ writeIORef jwkRef jwkset
+  !jwkset <- either (logAndThrow . parseErr) return $ J.eitherDecode respBody
```

Comment:
Might as well use `J.eitherDecode'` here if we want to be explicit about strictness, right?